jdaviz/configs/default/plugins/markers/markers.py

```diff
@@ -257,4 +391,55 @@ def _on_viewer_key_event(self, viewer, data):
             x, y = row_info['axes_x'], row_info['axes_y']
             self._get_mark(viewer).append_xy(getattr(x, 'value', x), getattr(y, 'value', y))
 
-            self.hub.broadcast(MarkersPluginUpdate(table_length=len(self.table), sender=self))
+        elif data['event'] == 'keydown' and data['key'] == 'r':
+            self.clear_viewer_marks()
+
+        elif data['event'] == 'keydown' and data.get('key') in ('d', '∂'):
+            if data.get('altKey', False):
+                coords = self._get_snap_coordinates(viewer)
+            else:
+                coords = self.coords_info.as_dict()
+            viewer_id = viewer.reference or viewer.reference_id
+
+            if self._distance_first_point is None:
+                self._distance_first_point = coords
+                self.distance_display = "..."
+            else:
+                p1 = self._distance_first_point
+                p2 = coords
+                self._distance_first_point = None
+
+                if self.app._align_by.lower() == 'wcs':
+                    world_avail = ('world_ra' in p1 and 'world_ra' in p2 and
+                                   p1.get('world_ra') is not None and p2.get('world_ra') is
+                                   not None)
+                    if world_avail:
+                        c1 = SkyCoord(p1['world_ra'], p1['world_dec'], unit='deg', frame='icrs')
+                        c2 = SkyCoord(p2['world_ra'], p2['world_dec'], unit='deg', frame='icrs')
+                        display_str = f"{c1.separation(c2).to(u.arcsec).value:.2f} arcsec"
+                    else:
+                        display_str = "N/A"

```

Comment:
is this possible?  If the app is aligned by WCS, then there has to be `world_avail`, right?  (See general comment about starting a line and then changing alignment type)